cmd/migration.go

```diff
@@ -1,131 +1,202 @@
 package cmd
 
 import (
+	"context"
+	"errors"
 	"fmt"
+
 	"github.com/MakeNowJust/heredoc"
-	guardianv1beta1 "github.com/odpf/guardian/api/proto/odpf/guardian/v1beta1"
+	. "github.com/odpf/guardian/api/proto/odpf/guardian/v1beta1"
 	"github.com/odpf/guardian/domain"
 	"github.com/odpf/guardian/internal/crypto"
-	"github.com/odpf/guardian/plugins/migrations/metabase"
+	"github.com/odpf/guardian/plugins/migrations"
+	mb "github.com/odpf/guardian/plugins/migrations/metabase"
 	"github.com/spf13/cobra"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func MigrationCmd() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:     "migration",
-		Aliases: []string{"migration"},
-		Short:   "Manage migration",
-		Long: heredoc.Doc(`
-			Work with migrations.
-
-			Migrations are used to populate guardian appeals from target system`),
-		Example: heredoc.Doc(`
-			$ guardian migration metabase
-		`),
-		Annotations: map[string]string{
-			"group:core": "true",
-		},
-	}
-
-	cmd.AddCommand(metabaseMigrationCmd())
-	bindFlagsFromConfig(cmd)
-
-	return cmd
-}
+const (
+	pending = "pending"
+	active  = "active"
+)
 
-func metabaseMigrationCmd() *cobra.Command {
-	return &cobra.Command{
-		Use:   "metabase",
-		Short: "Metabase migration",
+func MigrationCmd(config *Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Guardian migration",
```

Comment:
@rahmatrhd @singhvikash11 I think we need to think of a better approach than attaching it on migrate. One possible suggestion could be to call it import and scope it with the provider itself.

Reply:
@ravisuhag this is a good idea, we need to implement migrate for bigquery provider and other providers too. Then we could expose it as a sub-command on the provider

Reply:
@ravisuhag @rahmatrhd If you folks are okay with Metabase migrate for now then I can move this as sub-command to provider.

Reply:
@singhvikash11 What do you mean okay with metabase provider?

Also, i would recommend checking out https://github.com/GoogleCloudPlatform/terraformer for some inspiration on how command could look like.

Reply:
@ravisuhag we already have the command `guardian provider create` command to register the resources with guardian. Let's use the following command to import ACL of existing resources from a source system:

`guardian provider import <provide-name> acl `
`guardian provider import <provide-name> acl -r <resources>`

what do you guys think? cc @rahmatrhd

Reply:
@singhvikash11 What does `acl` represent in this command?

Reply:
@ravisuhag Bigquery, Metabase resources like table, dataset, collection etc defined by a set of permissions associated with a set of people, group, service-account. By importing access-control of these resources into our system we'll replicate it in the model of appeal-resource in Guardian.

Reply:
@singhvikash11 Yes, understood. But what i meant is that we don't need acl word in the comamnd right? the command can plainly be `guardian provider import <provide-name> -r <resources>` where `-r` flag will take the kind of resources you want to import from that provider.

Reply:
@ravisuhag This makes sense, we can go ahead with this command. 

cc @rahmatrhd